CheckPhoneNumberControl/CheckPhoneNumberControl/index.ts

```diff
@@ -170,13 +172,15 @@ export class CheckPhoneNumberControl implements ComponentFramework.StandardContr
 			}else {
 				this.handleValue(parsedPhoneNumber.getNumber(this._outputFormat));
 			}
-			this._notifyOutputChanged();
+			this._numberValidReturnValue = true;
 		}
 		else{
 			this._inputElement.classList.add("incorrect");
 			this._errorContainer.classList.add("inputError");
-			this._value = "";
+			this._value = this._inputElement.value;
+			this._numberValidReturnValue = false;
 		}
+		this._notifyOutputChanged();

```

Comment:
Why have you moved the line here?
My idea was that it should not be updated when the number is incorrect. Now the underlying field will be updated even when the number is incorrect.

Reply:
Yes, that's true. But the problem is, that we need to call notifyOutputChanged() for the "validity". If we do not do that, the validity field will not be updated.
I also don't see such a big problem here, because the Field is only updated on UI level.

Reply:
The problem could be if the "validity" field will not be used to check the form we could end up with wrong data in the field.
This can not be the case if we do not fill the value.

Maybe we should call the notify function but only set the real value when it is correct.

Reply:
Will be merged and issue will be created